### DISCUSSION TITLE
In [node]/child_process, stdout/stderr fields are missing on ExecException

### DISCUSSION THREAD

**Discussion by Thiht** [DefinitelyTyped/DefinitelyTyped#58854] [Issues with a @types package]
Hi,

It seems that the [ExecException](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/child_process.d.ts#L802-L807) interface lacks the `stdout` and `stderr` fields.

Here's a sample script showing that the fields do exist:

```js
import { promisify } from "util";
import { exec } from "child_process";

const execPromise = promisify(exec);

export const myCommand = async () => {
  const { stdout, stderr } = await execPromise(`false`);
  return {
    stdout,
    stderr,
  };
};

try {
  myCommand();
} catch (e) {
  console.log("Error", e);
}
```

Execution:

```
$ node test.mjs      
node:child_process:397
      ex = new Error('Command failed: ' + cmd + '\n' + stderr);
           ^

Error: Command failed: false

    at ChildProcess.exithandler (node:child_process:397:12)
    at ChildProcess.emit (node:events:390:28)
    at maybeClose (node:internal/child_process:1064:16)
    at Socket.<anonymous> (node:internal/child_process:450:11)
    at Socket.emit (node:events:390:28)
    at Pipe.<anonymous> (node:net:687:12) {
  killed: false,
  code: 1,
  signal: null,
  cmd: 'false',
  stdout: '',
  stderr: ''
}
```

This makes it really hard to retrieve the stdout/stderr in case of error.

I'm also not sure if it's the correct issue to ask it, but it seems impossible to do an instance check with the current implementation of ExecException:

```ts
try {}
catch (e) {
  if (e instanceof ExecException) {}
}
```

> 'ExecException' only refers to a type, but is being used as a value here.

[a comment by an automated account is not shown]

**Comment by thomaseizinger** [comment]
> I'm also not sure if it's the correct issue to ask it, but it seems impossible to do an instance check with the current implementation of ExecException:
> 
> ```ts
> try {}
> catch (e) {
>   if (e instanceof ExecException) {}
> }
> ```
> 
> > 'ExecException' only refers to a type, but is being used as a value here.

FWIW, the correct way to do this is to use a type-guard:


```ts
function isExecException(candidate: unknown): candidate is ExecException {
    if (candidate && typeof candidate === 'object' && 'cmd' in candidate) {
        return true;
    }
    return false;
}

try {
	// your code
} catch (e: unknown) {
    if (isExecException(e)) {

    }
}
```
